module/preference/PreferenceDialog.kt

```diff
@@ -403,12 +384,20 @@ object PreferenceDialog {
             override val preferences: List<PreferenceField> = listOf(autoSave)
 
             override fun submit() {
-                preferences.forEach { it.reset()  }; preferenceMgr.autoSave = autoSave.value
+                preferenceMgr.autoSave = autoSave.value
+                autoSave.modified = false
+            }
+
+            override fun reset() {
+                autoSave.value = preferenceMgr.autoSave
+                autoSave.modified = false
             }
         }
 
         class ProjectManager : PreferenceGroup(PROJECT_MANAGER) {
-            private val IGNORED_PATHS_PLACEHOLDER = ".git"
+            companion object {
+                private val IGNORED_PATHS_PLACEHOLDER = ".git"
```

Comment:
By constant I mean compile-time constant (where possible).

`const val` is a compile-time constant. `val` in a companion object is created (at least) once at runtime.

This change applies to the other placeholder too
```suggestion
                private const val IGNORED_PATHS_PLACEHOLDER = ".git"
```